src/ceph-daemon/ceph-daemon

```diff
@@ -280,6 +280,25 @@ def make_log_dir(fsid, uid=None, gid=None):
     makedirs(log_dir, uid, gid, LOG_DIR_MODE)
     return log_dir
 
+def move(src, dst):
+    """
+    Move a file from src to dst
+    """
+    if os.path.isdir(dst):
+        dst = os.path.join(dst, os.path.basename(src))
+    logger.debug('Move \'%s\' -> \'%s\'' % (src, dst))
+    try:
+        return shutil.move(src, dst)
+    except IOError:
+        if os.path.islink(src):
+            # shutil.move() in python2 does not handle symlinks correctly
+            logger.debug('symlink \'%s\' -> \'%s\'' % (os.readlink(src), dst))
+            os.symlink(os.readlink(src), dst)
+            os.unlink(src)
```

Comment:
If `unlink` raises, `symlink` already created the destination. Is there a way to make this a tiny bit more atomic without blowing up the complexity?

Reply:
I'm not sure of a better way, but I'm open to suggestions :smiley: